resources/js/bootstrap/stores/concept.js

```diff
@@ -0,0 +1,532 @@
+
+            const node = new Node({
+                ...concept,
+                tree: tgtTree,
+            });
+            return this.addConceptNode(node, tree);
+        },
+        // method to add newly created concepts to store
+        async addConcept(data, tree, broaderId) {
+            const concept = await addConcept(data, tree, broaderId);
+
+            const node = new Node({
+                ...concept,
+                tree: tree,
+            });
+            return this.addConceptNode(node, tree);
+        },
+        async toggleTopLevelState(id, tree) {
+            const data = await toggleTopLevelState(id, tree);
+            if(data.is_top_concept) {
+                this.handleAddRelation(-1, data.id, tree);
+            } else {
+                this.handleRemoveRelation(-1, data.id, tree);
+            }
+            return data;
+        },
+        async fetchConcept(id, tree) {
+            const concept = await fetchConcept(id, tree);
+            this.pushConcepts([concept], tree);
+        },
+        async fetchChildren(id, tree) {
+            tree = tree != 'sandbox' ? 'project' : tree;
+            const children = await fetchChildren(id, tree);
+            return this.pushConcepts(children, tree);
+        },
+        // method to add existing, fetched concepts to store
+        pushConcepts(concepts, tree) {
+            const nodes = [];
+            concepts.forEach(concept => {
+                const node = new Node({
+                    ...concept,
+                    tree: tree,
+                });
+                this.addConceptNode(node, tree, {
+                    ignore_count: true,
+                });
+                nodes.push(node);
+            });
+            return nodes;
+        },
+        async deleteConcept(id, tree, action, parameters) {
+            await deleteConcept(id, tree, action, parameters);
+
+            const conceptRefs = this.conceptReferences[tree][id];
+            const parentRefs = this.conceptParents[tree][id] || [];
+            // get all narrowers, simply get them from first ref
+            const conceptRef = conceptRefs[0];
+            const concept = this.conceptMap[tree][conceptRef];
+            if(action != '' && action != 'cascade') {
+                const narrowerIds = concept.narrowers.map(narrower => narrower.id);
+                let broaders = null;
+                if(action == 'level') {
+                    broaders = concept.is_top_concept ? [...parentRefs, -1] : parentRefs;
+                } else if(action == 'top') {
+                    broaders = [-1];
+                } else if(action == 'rerelate') {
+                    broaders = [parameters.p];
+                }
+                this.addRelation(broaders, narrowerIds, tree);
+            }
+
+            const removeBroaders = concept.is_top_concept ? [...parentRefs, -1] : parentRefs;
+            this.removeRelation(removeBroaders, [id], tree);
```

Comment:
The relation will be removed due to foreign key. When calling removeRelation it will try to remove the already deleted relation. It should be 
```js
           //// Relations will be automatically removed as they are defined as cascade delete 
            // this.removeRelation(id,removeBroaders, tree);
            this.handleRemoveRelation(removeBroaders, id, tree);         
```